extensions/stackdriver/log/exporter.cc

```diff
@@ -0,0 +1,102 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "extensions/stackdriver/log/exporter.h"
+
+#ifdef NULL_PLUGIN
+namespace Envoy {
+namespace Extensions {
+namespace Common {
+namespace Wasm {
+namespace Null {
+namespace Plugin {
+
+using envoy::api::v2::core::GrpcService;
+using Envoy::Extensions::Common::Wasm::Null::Plugin::GrpcStatus;
+using Envoy::Extensions::Common::Wasm::Null::Plugin::logDebug;
+using Envoy::Extensions::Common::Wasm::Null::Plugin::logInfo;
+using Envoy::Extensions::Common::Wasm::Null::Plugin::StringView;
+
+#endif
+
+constexpr char kGoogleStackdriverLoggingAddress[] = "logging.googleapis.com";
+constexpr char kGoogleLoggingService[] = "google.logging.v2.LoggingServiceV2";
+constexpr char kGoogleWriteLogEntriesMethod[] = "WriteLogEntries";
+constexpr char kDefaultRootCertFile[] = "/etc/ssl/certs/ca-certificates.crt";
+constexpr int kDefaultTimeoutMillisecond = 10000;
+
+namespace Extensions {
+namespace Stackdriver {
+namespace Log {
+
+ExporterImpl::ExporterImpl(RootContext* root_context,
+                           std::string logging_service_endpoint) {
+  context_ = root_context;
+  success_callback_ = [](google::protobuf::Empty&&) {
+    logDebug("successfully sent Stackdriver logging request");
+  };
+
+  failure_callback_ = [](GrpcStatus status, StringView message) {
+    // TODO: add retry.
+    logWarn("Stackdriver logging api call error: " +
+            std::to_string(static_cast<int>(status)) + std::string(message));
+  };
+
+  // Construct grpc_service for the Stackdriver gRPC call.
+  GrpcService grpc_service;
+  if (logging_service_endpoint.empty()) {
+    grpc_service.mutable_google_grpc()->set_target_uri(
+        kGoogleStackdriverLoggingAddress);
```

Comment:
Do we use Google style guide or Envoy style guide in Istio Proxy code? As Envoy suggests naming constants without k.. https://github.com/envoyproxy/envoy/blob/master/STYLE.md

Reply:
Hmm, yeah, but stackdriver filter starts with k convention. I think as long as we feel comfortable with it, it is fine to just use the google style?